modules/programs/hexchat.nix

```diff
@@ -0,0 +1,262 @@
+{ config, pkgs, lib, ... }:
+with builtins;
+with lib;
+let cfg = config.programs.hexchat;
+in {
+  meta.maintainers = [ maintainers.superherointj ];
+
+  options.programs.hexchat = {
+    channels = with lib.types;
+      let
+        modChannelOption = submodule {
+          options = {
+            autojoin = mkOption {
+              type = nullOr (listOf str);
+              description =
+                "Channels list to autojoin on connecting to server.";
+              example = ''[ "##linux" "#nix" ]'';
+              default = [ ];
+            };
+            charset = mkOption {
+              type = str;
+              description = "Charset";
+              default = "";
+              example = "UTF-8 (Unicode)";
+            };
+            commands = mkOption {
+              type = nullOr (listOf str);
+              description = "Commands to be executed on connecting to server.";
+              example = ''[ "ECHO Greetings fellow Nixer! ]'';
+              default = [ ];
+            };
+            loginMethod = mkOption {
+              type = with lib.types;
+                nullOr (types.enum [
+                  "nickServMsg"
+                  "nickServ"
+                  "challengeAuth"
+                  "sasl"
+                  "serverPassword"
+                  "saslExternal"
+                  "customCommands"
+                ]);
+              description = ''
+                null => Default
+                "nickServMsg" (1) => NickServ (/MSG NickServ + password)
+                "nickServ" (2) => NickServ (/NICKSERV + password)
+                "challengeAuth" (4) => Challenge Auth (username + password)
+                "sasl" (6) => SASL (username + password)
+                "serverPassword" (7) => Server password (/PASS password)
+                "saslExternal" (10) => SASL EXTERNAL (cert)
+                "customCommands" (9) => Custom... (connect commands)
+              '';
+              default = null;
+            };
+            nickname = mkOption {
+              type = str;
+              default = "";
```

Comment:
Do you need to set a default here? I think it would be better for it to break if the user didn't set it.

Reply:
HexChat suggests value when string is empty. It doesn't need to break.

Reply:
Default is now removed.